controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -455,17 +455,14 @@ func (r *BareMetalHostReconciler) actionInspecting(prov provisioner.Provisioner,
 		return recordActionFailure(info, metal3v1alpha1.InspectionError, provResult.ErrorMessage)
 	}
 
-	if details != nil {
-		info.host.Status.HardwareDetails = details
-		return actionComplete{}
-	}
+	info.host.ClearError()
```

Comment:
It looks like sometimes we clear the error in the controller and sometimes in the state machine. That may be an area to clean up in a future PR.

Reply:
The `ClearError` is invoked alongside to `actionComplete{}` or `actionContinue` (but not the viceversa). Could make sense to have a utility method to group those commands together, like the `recordActionFailure`?

Reply:
IIRC I looked into doing the `ClearError()` in a single place in `Reconcile()` when we get the actionResult. However, it's not as consistent as you would hope so we have to keep doing it on a case-by-case basis.

The changes in this last commit are just refactoring to make the control flow of this action look like the others, so I wouldn't get too hung up on this particular part. If I'd known this PR was going to take a year I'd have just submitted this commit separately.